### ISSUE TITLE
sst.Api does not support custom domain with a provided hosted zone

### ISSUE THREAD

**Issue by kamsar** [serverless-stack/serverless-stack#1771]
I'm reading the exports of a shared stack which deploys a Route53 hosted zone and trying to use it to create a `sst.Api` with a custom domain on that hosted zone.

```ts
const hostedZone: IHostedZone;
const certificate: ICertificate;

// critically, the zoneName is still a token at this point
const domainName = `${scope.stage}-api.${hostedZone.zoneName}`;

const api = new sst.Api(this, 'Api', {
    customDomain: {
      domainName,
      cdk: { hostedZone, certificate },
    },
  });
```

When I deploy this I get this error:

`Error: You also need to specify the "hostedZone" if the "domainName" is passed in as a reference.`

If I specify the `hostedZone` as a string, it complains because `hostedZone.zoneName` is still a token, but it's doing this check _without_ checking if `customDomain.cdk.hostedZone` is set, so it ignores that I already have it resolved.

https://github.com/serverless-stack/serverless-stack/blob/532b17aa5060b2eb1b2acc95b75fb3378ebe1b05/packages/resources/src/util/apiGatewayV2Domain.ts#L114

Seems like this should read like this instead:

```ts
if (cdk.Token.isUnresolved(customDomain.domainName)) {
    if (!(customDomain.hostedZone || customDomain.cdk.hostedZone)) {
      throw new Error(
        `You also need to specify the "hostedZone" if the "domainName" is passed in as a reference.`
      );
    }
  } else {
    assertDomainNameIsLowerCase(customDomain.domainName as string);
  }
  ```

**Comment by kamsar** [comment]
Can confirm, making the above change to the node_modules version fixed my issue.

**Comment by fwang** [comment]
Hey @kamsar, thanks for submitting the PR! It is released in v1.2.18.
